src/otg/common/spark_helpers.py

```diff
@@ -153,6 +192,38 @@ def calculate_neglog_pvalue(
     return -1 * (f.log10(p_value_mantissa) + p_value_exponent)
 
 
+def parse_pvalue(p_value: Column) -> tuple:
```

Comment:
Can we deal with p-values below 2.225E-308?

```
>>> import pyspark.sql.types as t
>>> d = [(1.0), (0.5), (1e-20), (3e-3), (1e-1000)]
>>> spark.createDataFrame(d, t.FloatType()).select('value',*parse_pvalue(f.col('value'))).show()
+-------+--------------+--------------+
|  value|pValueMantissa|pValueExponent|
+-------+--------------+--------------+
|    1.0|           1.0|             0|
|    0.5|           5.0|            -1|
|1.0E-20|           1.0|           -20|
|  0.003|           3.0|            -3|
|    0.0|         2.225|          -308|
+-------+--------------+--------------+
```

Reply:
I am afraid it wont work as the p-value is cast into float on the harmonization pipeline, so the input tsv probably contains zeroes already. But I will need to double check with James. But anyway, even if the harmonization pipeline handles p-values as text, it might get zeroes from the providers. This is something we need to prepare ourselves.